subsys/bluetooth/Kconfig.logging

```diff
@@ -7,844 +7,206 @@ config BT_LOG
-	help
-	  Use this option to enable Provisioning device debug logs for the
-	  Bluetooth Mesh functionality.
-
-config BT_MESH_DEBUG_ACCESS
-	bool "[DEPRECATED] Access layer debug"
-	select DEPRECATED
-	help
-	  Use this option to enable Access layer and device composition
-	  related debug logs for Bluetooth Mesh.
-
-config BT_MESH_DEBUG_MODEL
-	bool "[DEPRECATED] Foundation model debug"
-	select DEPRECATED
-	help
-	  Use this option to enable debug logs for the Foundation
-	  Models.
-
-config BT_MESH_DEBUG_ADV
-	bool "[DEPRECATED] Advertising debug"
-	select DEPRECATED
-	help
-	  Use this option to enable advertising debug logs for
-	  the Bluetooth Mesh functionality.
-
-config BT_MESH_DEBUG_LOW_POWER
-	bool "[DEPRECATED] Low Power debug"
-	select DEPRECATED
-	help
-	  Use this option to enable Low Power debug logs for the
-	  Bluetooth Mesh functionality.
-
-config BT_MESH_DEBUG_FRIEND
-	bool "[DEPRECATED] Friend debug"
-	select DEPRECATED
-	help
-	  Use this option to enable Friend debug logs for the
-	  Bluetooth Mesh functionality.
-
-config BT_MESH_DEBUG_PROXY
-	bool "[DEPRECATED] Proxy debug"
-	select DEPRECATED
-	depends on BT_MESH_GATT
-	help
-	  Use this option to enable Proxy protocol debug logs.
-
-config BT_MESH_DEBUG_SETTINGS
-	bool "[DEPRECATED] Persistent settings debug"
-	select DEPRECATED
-	depends on BT_SETTINGS
-	help
-	  Use this option to enable persistent settings debug logs.
-
-config BT_MESH_DEBUG_CDB
-	bool "[DEPRECATED] Configuration database debug"
-	select DEPRECATED
-	depends on BT_MESH_CDB
-	help
-	  Use this option to enable configuration database debug logs.
-
-config BT_MESH_DEBUG_CFG
-	bool "[DEPRECATED] Configuration debug"
-	select DEPRECATED
-	help
-	  Use this option to enable node configuration debug logs for the
-	  Bluetooth Mesh functionality.
-
-endmenu # [DEPRECATED] Mesh
-
-menu "[DEPRECATED] Services"
-
-# IAS
-
-config BT_DEBUG_IAS_CLIENT
-	bool "[DEPRECATED] Immediate Alert Service Client debug"
-	select DEPRECATED
-	depends on BT_IAS_CLIENT
-	help
-	  This option enables enables Immediate Alert Service Client debug logs.
-
-# OTS
-
-# TODO: Unify logging for OTS server and client - the client uses
-# "BT" debugging, the server does not.
-
-config BT_DEBUG_OTS_CLIENT
-	bool "[DEPRECATED] Object Transfer Service Client debug"
-	select DEPRECATED
-	help
-	  Use this option to enable Object Transfer Client debug logs
-
-endmenu # [DEPRECATED] Services
-
-endmenu # Bluetooth legacy logging options
-
-endif # BT_LOG_LEGACY
-
 # (subsys/bluetooth/Kconfig)
 
 module = BT
 module-str = "Bluetooth"
 source "subsys/logging/Kconfig.template.log_config"
 
+# Set BT as the parent module for all the symbols that will use
+# `Kconfig.template.log_config_inherit`. This is more convenient to do it here
+# instead of setting it for every modules.
+#
+# The downside of doing that is that people need to be careful when adding a new
+# parent module because  all the following symbol will use it.
+parent-module = BT
```

Comment:
I think it's fine to keep it. People adding new module don't have to worry about it at least

Reply:
Probably not, but this is out of scope for this PR. We would have to deprecate `CONFIG_BT_LOG_LEVEL` first.

Reply:
> We would have to deprecate CONFIG_BT_LOG_LEVEL first.

Not sure that it's true, `CONFIG_BT_LOG_LEVEL` is defined here: https://github.com/zephyrproject-rtos/zephyr/blob/a332064e1e4299f6789939520fe7dd2bd6aa40d1/subsys/bluetooth/Kconfig.logging#L625

Reply:
I assume @Thalley meant that inheritance from `CONFIG_BT_LOG_LEVEL` is not useful. But that is the current behavior, so it would have to be deprecated. Once that's deprecated, what purpose does `CONFIG_BT_LOG_LEVEL` serve? So I assumed we are talking about removing it.

Reply:
I agree that it's out of the scope to modify that part :)

Reply:
Anyway I think it's nice to have it, especially if there is update of the logging subsystem that use that